### DISCUSSION TITLE
Programmatically clean up after job completion

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8513] [Q&A]
Do Dagster jobs have standard hooks for post-run clean-up, i.e. a "finally" block kind of facility?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1636662400149500?thread_ts=1636662400.149500&cid=C01U954MEER).*

**Comment by yuhan** [comment]
We don't have anything that exactly matches that at the moment, but if you're looking for things like deleting temporary files / tearing down resources, we do have a concept of a [run_status_sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors), which can be tuned to either run failures, run successes, or other run statues.
